src/main/java/org/jabref/gui/externalfiles/FindFullTextAction.java

```diff
@@ -71,25 +73,34 @@ public void execute() {
         }
 
         Task<Map<BibEntry, Optional<URL>>> findFullTextsTask = new Task<Map<BibEntry, Optional<URL>>>() {
+
             @Override
             protected Map<BibEntry, Optional<URL>> call() {
                 Map<BibEntry, Optional<URL>> downloads = new ConcurrentHashMap<>();
                 int count = 0;
-                for (BibEntry entry : basePanel.getSelectedEntries()) {
-                    FulltextFetchers fetchers = new FulltextFetchers(Globals.prefs.getImportFormatPreferences());
-                    downloads.put(entry, fetchers.findFullTextPDF(entry));
+                AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(basePanel.getBibDatabaseContext(), Globals.prefs.getFilePreferences(), Globals.prefs.getAutoLinkPreferences(), ExternalFileTypes.getInstance());
+                List<BibEntry> changed = util.linkAssociatedFiles(basePanel.getSelectedEntries(), new NamedCompound(""));
+
+                for (BibEntry entry : changed) {
+                    if (entry.getFiles().isEmpty()) {
+                        FulltextFetchers fetchers = new FulltextFetchers(Globals.prefs.getImportFormatPreferences());
+                        downloads.put(entry, fetchers.findFullTextPDF(entry));
+                    } else {
+                        downloads.put(entry, Optional.empty());
```

Comment:
Why is an empty download task created? - Can't this line just be omitted?

Reply:
It's not an entry download task, it will never be downloaded and the findFullText only returns an URL.
 The onSucess method however iterates through the map and the real download will happen in addLinkedFileFromURL, if an URL is present

Reply:
This is a really ugly hack. Please clearly separate the two steps: first try to find unlinked files for all selected files and then download full text pdfs for the remaining entries.

Reply:
Separated the steps now.